app/Console/Commands/SupportEvents.php

```diff
@@ -0,0 +1,197 @@
+        ],
+        'fno_or_live_only' => [
+            // KZMA
+            'KZMA' => ['KAGR', 'KAPF', 'KAVO', 'KBCT', 'KBOW', 'KCHN', 'KCOI', 'KEYW', 'KFLL', 'KFMY', 'KFPR', 'KFXE', 'KGIF', 'KHST', 'KHWO', 'KIMM', 'KLAL', 'KLNA', 'KMCF', 'KMIA', 'KMKY', 'KMTH', 'KNQX', 'KOBE', 'KOPF', 'KPBI', 'KPCM', 'KPGD', 'KPHK', 'KPIE', 'KPMP', 'KRSW', 'KSEF', 'KSPG', 'KSRQ', 'KSUA', 'KTBW', 'KTMB', 'KTNT', 'KTPA', 'KTPF', 'KVNC', 'KVRB', 'KXMR', 'MBAC', 'MBNC', 'MBPI', 'MBSC', 'MBSY'],
+            // KZNY
+            'KZNY' => ['KABE', 'KAVP', 'KBDR', 'KBGM', 'KBLM', 'KCDW', 'KCKZ', 'KCTO', 'KCXY', 'KCZG', 'KDMW', 'KDXR', 'KDYL', 'KELM', 'KEVY', 'KEWR', 'KFOK', 'KFRG', 'KFWN', 'KHPN', 'KHTO', 'KHVN', 'KHWV', 'KHZL', 'KILG', 'KIPT', 'KISP', 'KJFK', 'KJRB', 'KLDJ', 'KLGA', 'KLHV', 'KLNS', 'KLOM', 'KMDT', 'KMGJ', 'KMJX', 'KMMU', 'KMPO', 'KMQS', 'KMUI', 'KNEL', 'KOQN', 'KOXC', 'KPHL', 'KPNE', 'KPOU', 'KPSB', 'KPTW', 'KRDG', 'KRVL', 'KSEG', 'KSMQ', 'KSWF', 'KTEB', 'KTHV', 'KTTN', 'KUKT', 'KUNV', 'KVAY', 'KWBW', 'KWRI', 'KXLL', 'KZER']
+        ]
+    ];
+
+    protected array $event_position_preset = [
+        "STBY | Standby",
+        "ZTL | Atlanta Center",
+        "A80 | Sattelite ATCT",
+        "A80 | Atlanta TRACON",
+        "ATL | Atlanta ATCT",
+        "CLT | Satellite ATCT",
+        "CLT | Charlotte TRACON",
+        "CLT | Charlotte ATCT",
+        "CIC/TMU",
+    ];
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle() {
+        $this->info("-- Updating support events - this may take a while --");
+
+        $this->info("Building ARTCC LUTs");
+
+        $all_events_lut = [];
+        $fno_or_live_only_lut = [];
+        $inverse_lut = [];
+
+        foreach ($this->event_pull_lut["all_events"] as $artcc => $airports) {
+            foreach ($airports as $airport) {
+                $all_events_lut[] = $airport;
+                $inverse_lut[$airport] = $artcc;
+            }
+        }
+        foreach ($this->event_pull_lut["fno_or_live_only"] as $artcc => $airports) {
+            foreach ($airports as $airport) {
+                $fno_or_live_only_lut[] = $airport;
+                $inverse_lut[$airport] = $artcc;
+            }
+        }
+
+
+        $client = new Client();
+
+        $this->info('Downloading event info from VATSIM');
+
+        $res = $client->get('https://my.vatsim.net/api/v1/events/all');
+
+        $this->info('Parsing event info JSON');
+
+        $result = json_decode($res->getBody());
+        if (is_null($result)) {
+            return;
+        }
+
+        foreach ($result->data as $event) {
+            $existing = Event::where('vatsim_id', $event->id)->first();
```

Comment:
Will this catch events that we (ZTL) have scheduled and possibly already created? Or do we have this to skip all ZTL events?

Reply:
This check is intended to prevent us from constantly creating the same event in our database over and over. When an event is created from the myVATSIM calendar, the `vatsim_id` field is set to the myVATSIM event ID. Our events will be filtered out by the airport check later down in the file - only airports for the facilities we want to pull from are included and thus our events will be skipped by that check.